### PR TITLE
Login

### PR DESCRIPTION
**Pull request by imstar15**
Call loginWithSignature to login

### CODE REVIEW

File: src/pages/TestAccount/TestAccount.js [hunk 1]
```diff
@@ -18,6 +19,20 @@ const TestAccount = () => {
     return signatureVerify(signedMessage, signature, hexPublicKey).isValid;
   };
 
+  const login = async () => {
+    const username = '53d8NKF@example.com';
+    const signinMessage = getSigninMessage();
+    const signature = await sign(signinMessage);
+    const { sessionToken } = await Parse.Cloud.run("loginWithSignature", { username, signature });
```
Comment (imstar15) [inline src/pages/TestAccount/TestAccount.js]: Call loginWithSignature

File: src/pages/TestAccount/TestAccount.js [hunk 2]
```diff
@@ -18,6 +19,20 @@ const TestAccount = () => {
...
+    const user = await Parse.User.become(sessionToken)
```
Comment (imstar15) [inline src/pages/TestAccount/TestAccount.js]: Call Parse.User.become with sessionToken

File: src/pages/TestAccount/TestAccount.js [hunk 3]
```diff
@@ -18,6 +19,20 @@ const TestAccount = () => {
...
+    console.log('user: ', user);
```
Comment (imstar15) [inline src/pages/TestAccount/TestAccount.js]:
```
{
    "username": "53d8NKF@example.com",
    "email": "53d8NKF@example.com",
    "createdAt": "2021-12-28T07:11:17.946Z",
    "sessionToken": "r:93c7b7117ea774a2f264ae53b9950123",
    "updatedAt": "2021-12-28T07:11:17.946Z",
    "ACL": {
        "*": {
            "read": true
        },
        "M49QO8mOk4": {
            "read": true,
            "write": true
        }
    },
    "__type": "Object",
    "className": "_User",
    "objectId": "M49QO8mOk4"
}
```

**Review by chrisli30 — APPROVED** [review]
Looks good to me, but also include normal login() in the code to make sure it fails with password.

File: src/pages/TestAccount/TestAccount.js [hunk 4]
```diff
@@ -18,6 +21,19 @@ const TestAccount = () => {
     return signatureVerify(signedMessage, signature, hexPublicKey).isValid;
   };
 
+  const login = async () => {
+    const signinMessage = getSigninMessage(username);
+    const signature = await sign(signinMessage);
+    const result = await Parse.User.logIn(username, password, { installationId: JSON.stringify({ signature, address: account }) });
```
Comment (chrisli30) [inline src/pages/TestAccount/TestAccount.js:27]: I don’t see a JSON.parse in parse backend code. We don’t need it when read the params on the backend?

File: src/pages/TestAccount/TestAccount.js [hunk 5]
```diff
@@ -33,19 +49,15 @@ const TestAccount = () => {
...
-    const username = email;
-
-    // Get signin_message from backend
-    const { signin_message: signinMessage } = await Parse.Cloud.run("getSigninMessage", { username, account });
-
-    // Sign In
+  const signUp = async () => {
+    if (Parse.User.current()){
+      await Parse.User.logOut();
+    }
```
Comment (chrisli30) [inline src/pages/TestAccount/TestAccount.js:60]: 👍

**Review by chrisli30 — APPROVED** [review]
Looks good to me!

**Pull request merged**